Add query using modification counter

rickpasetto opened the pull request:
This is a follow-up to #89 to add a query using the `modificationCounter`.  Hopefully this will help @darinkrauss with uploading to Tidepool Service.

rickpasetto committed:
Add query using modification counter

ps2 reviewed (approved):
LGTM! 🚀

darinkrauss commented on Loop/Managers/Alerts/AlertStore.swift:
```diff
@@ -128,6 +128,66 @@ extension AlertStore {
+    }
+    typealias QueryResult = Result<(QueryAnchor, [StoredAlert]), Error>
+
+    func executeAlertQuery(fromQueryAnchor queryAnchor: QueryAnchor? = nil, limit: Int = 100, completion: @escaping (QueryResult) -> Void) {
```
[nit] I didn't include defaults on the other store versions of execute*Query because I wanted any caller to explicitly set the `queryAnchor` (with full understanding what `nil` means and that it is only used in one particular case, that of the very first query) and `limit` (without indicating that any default size was preferred as it entirely depends upon the caller's intention).

rickpasetto replied:
IMHO, I think having a `nil` default for `queryAnchor` makes sense, as "start from the top by default" seems legit to me.
I do agree, however, that having a default `limit` of 100 is arbitrary and might be surprising.  I'll get rid of that default.

darinkrauss reviewed (approved):
LGTM! 🥧

rickpasetto committed:
Merge branch 'rick/LOOP-1200-alert-persistence' into rick/LOOP-1200-alert-persistence-query-support

rickpasetto committed:
PR Feedback

rickpasetto committed:
LOOP-1200: Dump Alert database into Issue Report (#93)

* LOOP-1200: Dump Alert database into Issue Report

Added query by date, plus unit tests.

* PR Feedback, plus some rework/refactor

This takes @darin's suggestion, with some modifications :).  Now, the Filter is part of the Anchor, which makes it so anchoring continues to apply the filter as appropriate.
